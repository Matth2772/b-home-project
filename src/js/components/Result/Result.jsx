import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import PropTypes from 'prop-types';

function Result(props) {
  const { result } = props;
  return (
    <ListGroup.Item>
      <a href={result.url}>
        {result.title}
      </a>
      {' '}
      -
      {' '}
      {result.author}
      {' '}
      -
      {' '}
      {result.points}
    </ListGroup.Item>
  );
}

Result.propTypes = {
  result: PropTypes.shape({
    title: PropTypes.string,
    author: PropTypes.string,
    points: PropTypes.number,
    url: PropTypes.string,
  }),
};

Result.defaultProps = {
  result: {
    title: '',
    author: '',
    points: '',
    url: '',
  },
};

export default Result;
