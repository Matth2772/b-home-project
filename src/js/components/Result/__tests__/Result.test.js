/* eslint-disable */
import React from 'react';
import Result from '../Result';
import renderer from 'react-test-renderer';

test('Result rendering well', () => {
  const component = renderer.create(
    <Result/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
