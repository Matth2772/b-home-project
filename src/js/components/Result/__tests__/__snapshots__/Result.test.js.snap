// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result rendering well 1`] = `
<div
  className="list-group-item"
  disabled={false}
  onClick={[Function]}
>
  <a
    href=""
  >
    
  </a>
   
  -
   
  
   
  -
   
  0
</div>
`;
